### ISSUE TITLE
is device mobile or not?

### ISSUE THREAD

**Issue by ezl** [ezl/dailyidea.com#480]
at several places in the app, we may be asking whether a device is a mobile device or not in order to decide what to show a user (draggable/swipe UI, should we show mobile login screen, anything else in the future)

store this logic somewhere that other parts can reuse it.

(@f-prime you may already have this done inside the swipeable logic -- if so, let's just take that and refactor for reusability)

@alisonuxdesign cc'ing, no input required.

**Comment by f-prime** [comment]
I want to post this here because from some other issues it seems like there have been a few ideas being tossed around for detecting mobile. I didn't have to solve this problem for swiping, but I did have to solve it for native share.

We can easily determine if the user is on mobile by checking if `navigator.share` is defined or not. This is how we access the native share API and it is undefined in Desktop browser. This seems to be the way to do it with the least amount of code. I have confirmed that it works on both Android and iOS.

Here is where I am using it in MenuPanel.vue. We can extract it out.

https://github.com/ezl/dailyidea.com/blob/40f32fe34614ca69106bc6fc385650d8a58dba33/front/components/ideaDetail/MenuPanel.vue#L194-L203
